### ISSUE TITLE
Multiple consumers problem, only first one work!

### ISSUE THREAD

**Issue by Mehrdad-Dadkhah** [segmentio/kafka-go#1031]
Hi
I run 2 consumers with same `Topic` and `GroupID`, but only one of them works and second one only wait for message and no message read! I have lots of messages ....

**Kafka Version**

> * Kafka version: 3.2
> * kafka-go version: 0.4.38

```go

func GetReader(topic string) *kafka.Reader {
	// make a new reader that consumes from topic-A, partition 0, at offset 42
	r := kafka.NewReader(kafka.ReaderConfig{
		Brokers: []string{
			configs.KAFKA_BOKER1_HOST,
		},
		Topic:     topic,
		Partition: 0,
		GroupID:   strings.ToUpper(fmt.Sprintf("%s-consumers", topic)),
		MinBytes:  10e3, // 10KB
		MaxBytes:  10e6, // 10MB
		ErrorLogger: kafka.LoggerFunc(func(msg string, args ...interface{}) {
			lctx := logger.Log.With()
			for k, a := range args {
				lctx.Str(
					fmt.Sprintf("kafak-arg-%v", k),
					dump.Format(a),
				)
			}

			lg := logger.GetLogger(lctx)
			err := errors.New(msg)
			lg.Error().AnErr("[KAFKA] [CONSUME]", err).Msg("")
		}),
	})

	return r
}

func ConsumeMessage(topic string, callback func(value []byte) error) {

	reader := GetReader(topic)
	m, err := reader.ReadMessage(context.Background())
	if err != nil {
		logger.Log.Error().AnErr("[KAFKA] [CONSUME]", err).Msg("read message error")
		panic(err)
	}
	logger.Log.Info().Msg(fmt.Sprintf("message at topic:%v partition:%v offset:%v	%s", m.Topic, m.Partition, m.Offset, string(m.Key)))

	callback(m.Value)

	err = reader.CommitMessages(context.Background(), m)
	if err != nil {
		logger.Log.Error().AnErr("[KAFKA] [CONSUME] [COMMIT]", err).Msg("commit message error")
	}
}
```

**Comment by maksim77** [comment]
It seems to me that when creating Reader, you specify both Partition and GroupID at the same time. According to the library documentation, you have to specify one thing.

From Kafka's point of view, this is also logical. Either a group of consumers that is balanced by Kafka itself, or you explicitly specify which partition to read.

Also if you use the ReadMessage method there is no need to commit the offset. Also if you use the ReadMessage method there is no need to commit the offset.

**Comment by dominicbarnes** [comment]
I also noticed the `Partition: 0` in your code, which is a little peculiar, but 0 is the default value so this likely isn't a problem. (I believe the Reader would error during init if it got confusing code)

I did notice that you're using `ReadMessage` along with `CommitMessages`, which is not how these APIs are designed to be used. `ReadMessage` automatically fetches a message and commits it's offset immediately, so if you want to manage offsets yourself, [use `FetchMessage` and `CommitMessages` instead](https://github.com/segmentio/kafka-go#explicit-commits).

I wonder if this usage of `CommitMessages` is colliding with what `ReadMessage` is already doing behind the scenes, this could cause problems for you.

tl;dr: try using either `ReadMessage` **only**, or `FetchMessage` and `CommitMessages` together, don't mix-and-match.

**Comment by nobbynobbs** [comment]
How many partitions do you have in the kafka topic?

**Comment by maksim77** [comment]
> How many partitions do you have in the kafka topic? 

But it looks like the reason! If the author of the question has one partition, then the behavior is absolutely normal.

**Comment by yaziedda** [comment]
set partition > 1 it's work for me

**Comment by dominicbarnes** [comment]
Due to lack of response, I'm going to close this issue out. Please re-open if you have more information to provide. :)
